components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -83,10 +90,67 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
     }
 
     pub fn propose_pending_command<T>(&mut self, ctx: &mut StoreContext<EK, ER, T>) {
-        if let Some(encoder) = self.raw_write_encoder_mut().take() {
+        if let Some(encoder) = self.simple_write_encoder_mut().take() {
             let (data, chs) = encoder.encode();
             let res = self.propose(ctx, data);
             self.post_propose_write(ctx, res, chs);
         }
     }
 }
+
+impl<EK: KvEngine, R> Apply<EK, R> {
+    #[inline]
+    pub fn apply_put(&mut self, cf: &str, key: &[u8], value: &[u8]) -> Result<()> {
```

Comment:
It seems missing relevant metrics such as lock_cf_written_bytes, size_diff_hint, etc.

Reply:
There may be too many TODOs. Metrics will be added all together in a separate PR.